src/components/Data/Browse/IndexBranch.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+  <div v-bind:class="{ 'open': open }">
+    <i v-if="collectionCount(index)" class="fa fa-caret-right tree-toggle" aria-hidden="true" @click="toggleBranch()"></i>
+    <!-- v-link="{name: 'DataIndex', params: {index: index.name}}" -->
+    <a class="tree-item truncate"
+       v-bind:class="{ 'active': isIndexActive($route, index.name) }">
+      <i class="fa fa-database" aria-hidden="true"></i>
+      <strong>{{index.name}}</strong> ({{collectionCount(index)}})
+    </a>
+    <ul class="collections">
+      <li v-for="collection in index.collections.stored">
+        <a class="tree-item truncate"
+           v-link="{name: 'DataIndexCollection', params: {index: index.name, collection: collection}}"
+           v-bind:class="{ 'active': isCollectionActive($route, collection) }">
+           <i class="fa fa-th-list" aria-hidden="true" title="Persisted collection"></i>
+           {{collection}}
+         </a>
+      </li>
+      <li v-for="collection in index.collections.realtime">
+        <a class="tree-item truncate">
+          <i class="fa fa-bolt" aria-hidden="true" title="Volatile collection"></i>
+          {{collection}}
+        </a>
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+export default {
+  props: ['index'],
+  data: function () {
+    return {
+      open: false
+    }
+  },
+  methods: {
+    toggleBranch () {
+      this.open = !this.open
+    },
+    collectionCount (index) {
+      let count = 0
+
+      if (!index.collections) {
+        return 0
+      }
+
+      if (index.collections.stored) {
+        count += index.collections.stored.length
+      }
+
+      if (index.collections.realtime) {
+        count += index.collections.realtime.length
+      }
+
+      return count
+    },
+    isIndexActive (routeObject, indexName) {
```

Comment:
I think we have to do other checks here:
 1. if the index or collection is selected, open the tab (that's ok)
 2. if the user has open or close a section we have to let into this state